flyteplugins/go/tasks/plugins/k8s/pod/plugin.go

```diff
@@ -236,15 +236,9 @@ func (plugin) GetTaskPhaseWithLogs(ctx context.Context, pluginContext k8s.Plugin
 
 	if err != nil {
 		return pluginsCore.PhaseInfoUndefined, err
-	} else if phaseInfo.Phase() != pluginsCore.PhaseRunning && phaseInfo.Phase() == pluginState.Phase &&
```

Comment:
This logic moved without change to `k8s.MaybeUpdatePhaseVersion(` so that it can be reused in other plugins.